packages/template-phpstan-compiler/config/php-parser.neon

```diff
@@ -1,11 +1,28 @@
+# see original config.neon in phpstan.neon - https://github.com/phpstan/phpstan-src/blob/386eb913abb6ac05886c5642fd48b5d99db66a20/conf/config.neon#L1582
+# this file overrides definitions from the config above
 services:
     defaultAnalysisParser:
-        factory: @cachedCurrentPhpVersionRichParser
+        factory: @pathRoutingParser
         arguments!: []
 
+    pathRoutingParser:
+        class: PHPStan\Parser\PathRoutingParser
+        arguments:
+            currentPhpVersionRichParser: @rectorParser
+            currentPhpVersionSimpleParser: @rectorParser
+            php8Parser: @php8Parser
+        autowired: false
+
+    rectorParser:
+        class: PHPStan\Parser\RichParser
+        arguments:
+            parser: @currentPhpVersionPhpParser
+            lexer: @currentPhpVersionLexer
+        autowired: false
```

Comment:
This is needed, so PHPStan fixtures for PHP 8.1 are parsed by PHP 8.1 parser, while code is parsed with "current version parser".
After upgrade, this was broken, as all code was parsed by PHP 8.0 parser. The fixtures were crashing on syntax error then.

This fixes it :+1: